plugin.js

```diff
@@ -86,9 +90,12 @@ function checkVersion (version, pluginName) {
   var fastifyVersion
   try {
     var pkgPath
-    if (require.main.filename) {
+    if (require.main && require.main.filename) {
       // We need to dynamically compute this to support yarn pnp
-      pkgPath = join(dirname(require.resolve('fastify', { paths: [require.main.filename] })), 'package.json')
+      pkgPath = resolvePkgPath(require.main.filename)
+    } else if (process.argv[1]) {
```

Comment:
In AWS Lambda this is causing the unwanted log statement "fastify not found, proceeding anyway"
In AWS Lambda the process.argv are: [ '/var/lang/bin/node', '/var/runtime/index.js' ]

@mcollina Do you want me to create a PR to check it like this?
```javascript
//...
} else if (process.argv[1] && process.argv[1] !== '/var/runtime/index.js' ) {
//...
```

Reply:
I don't think that's a correct fix. What's the `require.main` and `require.name.filename` in lambdas?
In other terms, I think your breakage was caused by https://github.com/fastify/fastify-plugin/commit/b3f4aa183e8a9ae6693cebee1c8b267c942b517e#diff-18e559ebd3cc204a52bc1dc3db2adf8b not this PR.

Can you open a new issue?

Reply:
You're probably right... will create a little console.log example...

Reply:
```
require.main.filename = /var/runtime/index.js

require.main = {
  id: '.',
  path: '/var/runtime',
  exports: {},
  parent: null,
  filename: '/var/runtime/index.js',
  loaded: true,
  children: [
    Module {
      id: '/var/runtime/RAPIDClient.js',
      path: '/var/runtime',
      exports: [Function: RAPIDClient],
      parent: [Circular],
      filename: '/var/runtime/RAPIDClient.js',
      loaded: true,
      children: [Array],
      paths: [Array]
    },
    Module {
      id: '/var/runtime/Runtime.js',
      path: '/var/runtime',
      exports: [Function: Runtime],
      parent: [Circular],
      filename: '/var/runtime/Runtime.js',
      loaded: true,
      children: [Array],
      paths: [Array]
    },
    Module {
      id: '/var/runtime/UserFunction.js',
      path: '/var/runtime',
      exports: [Object],
      parent: [Circular],
      filename: '/var/runtime/UserFunction.js',
      loaded: true,
      children: [Array],
      paths: [Array]
    },
    Module {
      id: '/var/runtime/Errors.js',
      path: '/var/runtime',
      exports: [Object],
      parent: [Module],
      filename: '/var/runtime/Errors.js',
      loaded: true,
      children: [],
      paths: [Array]
    },
    Module {
      id: '/var/runtime/BeforeExitListener.js',
      path: '/var/runtime',
      exports: [Object],
      parent: [Module],
      filename: '/var/runtime/BeforeExitListener.js',
      loaded: true,
      children: [],
      paths: [Array]
    },
    Module {
      id: '/var/runtime/LogPatch.js',
      path: '/var/runtime',
      exports: [Object],
      parent: [Module],
      filename: '/var/runtime/LogPatch.js',
      loaded: true,
      children: [],
      paths: [Array]
    }
  ],
  paths: [ '/var/runtime/node_modules', '/var/node_modules', '/node_modules' ]
}

process.argv = [ '/var/lang/bin/node', '/var/runtime/index.js' ]
```